src/Merger.hack

```diff
@@ -20,17 +22,17 @@ abstract final class Merger {
    *
    * In the case of duplicates, the last definition is used.
    */
-  public static function merge(\ConstVector<AutoloadMap> $maps): AutoloadMap {
+  public static function merge(vec<AutoloadMap> $maps): AutoloadMap {
     return shape(
-      'class' => self::mergeImpl($maps->map($map ==> $map['class'])),
-      'function' => self::mergeImpl($maps->map($map ==> $map['function'])),
-      'type' => self::mergeImpl($maps->map($map ==> $map['type'])),
-      'constant' => self::mergeImpl($maps->map($map ==> $map['constant'])),
+      'class' => self::mergeImpl(Vec\map($maps, $map ==> $map['class'])),
+      'function' => self::mergeImpl(Vec\map($maps, $map ==> $map['function'])),
+      'type' => self::mergeImpl(Vec\map($maps, $map ==> $map['type'])),
+      'constant' => self::mergeImpl(Vec\map($maps, $map ==> $map['constant'])),
     );
   }
 
   private static function mergeImpl(
-    Traversable<darray<string, string>> $maps,
+    Traversable<KeyedTraversable<string, string>> $maps,
   ): darray<string, string> {
```

Comment:
Not changing this to dict?

Reply:
In this branch, the AutoloadMap type is still dependent on darray-ness.
```HACK
type AutoloadMap = shape(
  'class' => darray<string, string>,
  'function' => darray<string, string>,
  'type' => darray<string, string>,
  'constant' => darray<string, string>,
);
```

lexidor#1 changes this to `dict<string, dict<string, string>>`.